scripts/user_retirement/utils/helpers.py

```diff
@@ -229,11 +165,19 @@ def _setup_all_apis_or_exit(fail_func, fail_code, config):
                 client_secret,
             )
 
-        if segment_base_url:
-            config['SEGMENT'] = SegmentApi(
-                segment_base_url,
-                segment_auth_token,
-                segment_workspace_slug
+        # Load and configure any retirement plugin APIs that are installed
+        for state in config['retirement_pipeline']:
+            service_name = state[2]
+
+            # Skip any states that have already loaded APIs
+            if service_name in config:
+                continue
+
+            mgr = driver.DriverManager(
+                namespace="retirement_driver",
+                name=service_name
             )
+            config[service_name] = mgr.driver.get_instance(config)
```

Comment:
These few lines are where the magic happens, if we don't already have an "API" configured (LMS, Ecomm, License Manager, etc would already be configured here) we take the value from our configuration and try to fetch a plugin that has a matching name in the `retirement_driver` entrypoint namespace. So `AMPLITUDE`, for instance.

If it exists, the given class will be have `get_instance(config)` called on it to return a fully configured instance of the class. If it does not exist currently this code will raise an error, if this passes we would probably want to wrap that in something with a much more clear error message.